.orchestra/config/components/openssl.yml

```diff
@@ -0,0 +1,53 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:overlay", "overlay")
+#@ load("/lib/make.lib.yml", "make")
+
+#@yaml/text-templated-strings
+---
+#@ def openssl_component():
+#@ source_url = "https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_1_1q.tar.gz"
+license: LICENSE
+builds:
+  default:
+    build_dependencies:
+      - host-c-toolchain
+      - glibc
+    configure: |
+      mkdir -p "$BUILD_DIR"
+      extract.sh --into "$BUILD_DIR" (@= source_url @)
+      cd "$BUILD_DIR" && ./Configure \
+        --prefix="$ORCHESTRA_ROOT" \
+        --openssldir="$ORCHESTRA_ROOT/etc/ssl" \
```

Comment:
Maybe we should use `/etc/ssl`. Not sure OpenSSL ships root CAs.
Also it's probably being hardcoded, if we keep it as is, we need to see if there's an environment variable to set to override it.

Reply:
I think in general it's indeed safest to use the distro/user provided certs anyway

Reply:
I changed it to `/etc/ssl` but now a question arises: what if the distro does not use `/etc/ssl`? I was thinking that we should instead provide our own fake config that tries to include all distro configs

Reply:
You can check `/etc/ssl` exists before invoking configure and error out if not.

Reply:
Yes but this won't prevent the user from installing it from a binary archive

Reply:
Ah eh, you're right. We'd need a post-unpack phase, which we don't have. Leave a comment.
Do you actually know a distro not using `/etc/ssl`?

Reply:
NixOS? Guix? Android? (though we'd have many more problems with Android)

I guess though that users of these distros would be used to troubleshooting problems caused by their usage of non-posix-compliant paths.